python_modules/libraries/dagster-dg/dagster_dg/cli/scaffold.py

```diff
@@ -276,23 +390,74 @@ def scaffold_github_actions_command(git_root: Optional[Path], **global_options:
     workflows_dir = git_root / ".github" / "workflows"
     workflows_dir.mkdir(parents=True, exist_ok=True)
 
-    template = SERVERLESS_GITHUB_ACTION_FILE.read_text()
-
     if plus_config and plus_config.organization:
         organization_name = plus_config.organization
         click.echo(f"Using organization name {organization_name} from Dagster Plus config.")
     else:
         organization_name = click.prompt("Dagster Plus organization name") or ""
 
+    if plus_config:
+        gql_client = DagsterPlusGraphQLClient.from_config(plus_config)
+
+        agent_type = _get_deployment_agent_type(gql_client)
+
+    else:
+        click.echo("No Dagster Plus config found.")
+        agent_type = click.prompt(
+            "Deployment agent type: ", type=click.Choice(("serverless", "hybrid"))
+        ).upper()
+
+    if agent_type == "SERVERLESS":
+        click.echo("Using serverless workflow template.")
+    else:
+        click.echo("Using hybrid workflow template.")
+
+    registry_url = None
+    additional_secrets_hints = []
+
+    template = (
+        SERVERLESS_GITHUB_ACTION_FILE.read_text()
+        if agent_type == "SERVERLESS"
+        else HYBRID_GITHUB_ACTION_FILE.read_text()
+    )
     template = template.replace("ORGANIZATION_NAME", organization_name)
 
+    if agent_type == "HYBRID":
+        # Attempt to read registry from registry.yaml
+        registry_file = dg_context.root_path / "registry.yaml"
+        if registry_file.exists():
+            registry_url = yaml.safe_load(registry_file.read_text()).get("registry")
+
+        if not registry_url:
+            registry_url = click.prompt("Docker registry URL, for built images")
+        else:
+            click.echo(f"Using Docker registry URL {registry_url} from registry.yaml")
+
+        project_contexts = _get_project_contexts(dg_context, cli_config)
+        for location_ctx in project_contexts:
+            _create_deploy_dockerfile(location_ctx.root_path / "Dockerfile", "3.11")
```

Comment:
Note that dg deploy also scaffolds a Dockerfile.. Lots of dockerfile scaffolding happening, should we have a separate command for it instead? maybe something that also scaffolds the build.yaml etc. that you would need to deploy either locally or via CI/CD?

Reply:
Moved to shared util, but follow up will add a separate command for this